agent/local/state.go

```diff
@@ -1079,7 +1079,7 @@ func (l *State) SyncChanges() error {
 			l.logger.Printf("[DEBUG] agent: Service %q in sync", id)
 		}
 		if err != nil {
-			return err
+			l.logger.Printf("[WARN] agent: Failed to register service %q: %s", id, err)
```

Comment:
It would be preferable if we could turn the error into a nil inside each of the sync*, delete* methods in the same way we do for v8 ACL permission failures.

That would mean this won't have any other unexpected changes of behaviour for totally unrelated issues that currently error here. I can't immediately think of a reason it would be dangerous not to stop here on an error although it might be pointless in many cases to keep trying RPCs while there is no connection to server or something. But the fact that we already special case some error conditions like ACL denied in those methods rant her than globally catching everything here increases the chance of breaking an assumption with this change.

Can you clarify what the error does look like in the old ACL that doesn't exist any more case? Is there a reliable way to sniff for it?

Reply:
The idea behind this was to prevent other cases when an error stops the whole process in the future. But I understand your point of view.
The condition catches `ErrDisabled` but not `ErrNotFound`, which is causing the issue.
I will change this to catch both errors and revert the continue on error in the loop.